src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -221,7 +221,7 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
             }
 
             var contentRootPath = ResolveContentRootPath(_options.ContentRootPath, AppContext.BaseDirectory);
-            var applicationName = _options.ApplicationName;
+            var applicationName = _options.ApplicationName ?? Assembly.GetEntryAssembly()?.GetName().Name;
```

Comment:
This does not guarantee to be not null. GetEntryAssembly() returns null when launching a net461 test project via xunit. Launching an net461 app by itself or launching a netcoreapp2.0 test project via xunit returns a valid value. The suspicion here is that launching via a different AppDomain in net461 doesn't resolve this correctly.